File: src/main/java/io/playground/orderservice/exception/BusinessErrorDto.java
package io.playground.orderservice.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class BusinessErrorDto {
    private final String code;
    private final String message;
    private final String detail;
    private final HttpStatus httpStatus;

    public BusinessErrorDto(BusinessErrorCode errorCode,
                            String detail) {
        this.code = errorCode.getCode();
        this.message = errorCode.getMessage();
        this.detail = detail;
        this.httpStatus = errorCode.getHttpStatus();
    }

    public static BusinessErrorDto from(BusinessDetailException e) {
        return new BusinessErrorDto(
                e.getErrorCode(),
                e.getDetail()
        );
    }
}

File: src/main/kotlin/io/playground/orderservice/exception/BusinessErrorDto.kt
package io.playground.orderservice.exception

import org.springframework.http.HttpStatus

class BusinessErrorDto(
    val code: String,
    val message: String,
    val detail: String?,
    val httpStatus: HttpStatus,
) {
    constructor(errorCode: BusinessErrorCode, detail: String?) : this(
        errorCode.code,
        errorCode.message,
        detail,
        errorCode.httpStatus,
    )

    companion object {
        @JvmStatic
        fun from(e: BusinessDetailException): BusinessErrorDto =
            BusinessErrorDto(e.errorCode, e.detail)
    }
}
